src/domain/event/eventForm/ValidationSchema.ts

```diff
@@ -35,9 +38,35 @@ const createValidationSchemaYup = (
       .typeError(VALIDATION_MESSAGE_KEYS.DATE)
       .required(VALIDATION_MESSAGE_KEYS.DATE_REQUIRED)
       .test(
-        'isTodayOrInTheFuture',
-        VALIDATION_MESSAGE_KEYS.DATE_TODAY_OR_LATER,
-        isTodayOrLater
+        'isInTheFuture',
+        VALIDATION_MESSAGE_KEYS.DATE_IN_THE_FUTURE,
+        isFuture
+      )
+      .when(
+        ['occurrenceDate', 'occurrenceStartsAt'],
+        (
+          occurenceDate: Date,
+          occurrenceStartsAt: string,
+          schema: Yup.DateSchema
+        ) => {
+          if (isValidDate(occurenceDate)) {
+            const isValid = isValidTime(occurrenceStartsAt);
+            const occurenceStart = isValid
+              ? parseDate(occurrenceStartsAt, 'HH:mm', occurenceDate)
+              : occurenceDate;
+            return schema.test(
+              'isBefore',
+              () => ({
+                key: VALIDATION_MESSAGE_KEYS.DATE_MAX,
+                max: formatDate(occurenceStart, DATETIME_FORMAT),
+              }),
+              (enrolmentStart: Date) => {
+                return enrolmentStart < occurenceStart;
+              }
```

Comment:
Very minor but could use implicit return

```
 (enrolmentStart: Date) => enrolmentStart < occurenceStart;
```

Reply:
fixed, although sometimes is easier to use longer format so it's easy to throw in console.log when debugging.

Reply:
one trick is to use comma operator for console.log:

`(enrolmentStart: Date) => (console.log(enrolmentStart), enrolmentStart < occurenceStart);`